schemas/sublime-project.json

```diff
@@ -66,13 +66,13 @@
       }
     },
     "settings": {
-      "type": "object"
+      "$ref": "sublime://schemas/preferences.sublime-settings"
```

Comment:
Does this also include the settings from `sublime-package.json` files?

Reply:
Yes and no.

Not automatically but it will work if you explicitly add them with something like:
```js
      {
        "file_patterns": [
          "/*.sublime-project"
        ],
        "schema": {
          "properties": {
            "settings": {
              "properties": {
                "LSP": {
                  "type": "object",
                  "markdownDescription": "The dictionary of your configured language servers or overrides for existing configurations. The keys of this dictionary are free-form. They give a humany-friendly name to the server configuration. They are shown in the bottom-left corner in the status bar once attached to a view (unless you have `\"show_view_status\"` set to `false`).",
                  "additionalProperties": {
                    "$ref": "sublime://settings/LSP#/definitions/ClientConfig"
                  },
                },
                "lsp_format_on_save": {
                  "$ref": "sublime://settings/LSP#/definitions/lsp_format_on_save",
                },
                "lsp_code_actions_on_save": {
                  "$ref": "sublime://settings/LSP#/definitions/lsp_code_actions_on_save"
                }
              }
            }
          }
        }
      }
```

So it requires a bit of duplication.

The bigger problem will be with syntax-specific settings since you can't know all the file patterns that should be extended (not without using the API) so it won't be that easy to make those work. Basically the problem is that schemas can only be merged by `file_patterns` and not by IDs. Or you can merge using `allOf` and co. but if multiple packages extend same schema this way then there is no way to get the "final" schema. It's a tricky problem that I'm not sure how to fix.